### PR TITLE
feat: image list used by KubeCF

### PR DESCRIPTION
**Pull request by f0rmiga**
## Description

Supersedes https://github.com/cloudfoundry-incubator/kubecf/pull/557.
https://github.com/cloudfoundry-incubator/kubecf/pull/557 was used as a base for the logic contained in this PR, although this PR improves the output and captures all combinations of feature flags, tests and non-BOSH releases.

After inspecting the list output, 2 BOSH releases not used were not removed from the cf-deployment manifest. This PR also adds ops-files to remove them (bosh-dns-aliases and bpm).

## Motivation and Context

Knowing all the images KubeCF is useful for caching, image copying, image inspection, etc.

## How Has This Been Tested?

Checking that `bazel build //deploy/helm/kubecf:image_list` produces `bazel-bin/deploy/helm/kubecf/image_list.json` and `bazel run //deploy/helm/kubecf:print_image_list` prints the artifact containing the image list to the terminal.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: deploy/helm/kubecf/BUILD.bazel [hunk 1]
```diff
@@ -75,3 +80,13 @@ helm_version(
     name = "version",
     chart = ":kubecf",
 )
+
+image_list(
```
Comment (viovanov) [inline deploy/helm/kubecf/BUILD.bazel]: Add some comments please for what these do.

File: deploy/helm/kubecf/BUILD.bazel [hunk 2]
```diff
@@ -75,3 +80,13 @@ helm_version(
...
+    name = "image_list",
+    chart = ":kubecf",
+)
+
+print_files(
```
Comment (viovanov) [inline deploy/helm/kubecf/BUILD.bazel]: Add some comments please for what these do.

File: deploy/helm/kubecf/assets/operations/releases.yaml [hunk 3]
```diff
@@ -1,3 +1,10 @@
+# Remove unused releases from cf-deployment.
```
Comment (viovanov) [inline deploy/helm/kubecf/assets/operations/releases.yaml]: 👍

File: rules/ioutil/def.bzl [hunk 4]
```diff
@@ -0,0 +1,23 @@
+"""
+This Bazel extension contains the set of rule definitions to deal with generic IO.
+"""
+
+def _print_files_impl(ctx):
```
Comment (viovanov) [inline rules/ioutil/def.bzl]: Please comment this as well.
Is it possible to lint these files?

File: rules/kubecf/def.bzl [hunk 5]
```diff
@@ -0,0 +1,70 @@
+"""
```
Comment (viovanov) [inline rules/kubecf/def.bzl]: Needs comments for everything

Comment (f0rmiga) [reply rules/kubecf/def.bzl]: Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/571/commits/73796132ab74bf44ac495e988e2cc4ce5f8d5e14.

File: rules/ioutil/def.bzl (hunk 4 above)
Comment (f0rmiga) [reply rules/ioutil/def.bzl]: Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/571/commits/73796132ab74bf44ac495e988e2cc4ce5f8d5e14.

Yes, it's possible to use the `buildifier` linter.

File: deploy/helm/kubecf/BUILD.bazel (hunk 1 above)
Comment (f0rmiga) [reply deploy/helm/kubecf/BUILD.bazel]: Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/571/commits/5fcab07449d5553739b0266bce81a105f48fa201.

File: deploy/helm/kubecf/BUILD.bazel (hunk 2 above)
Comment (f0rmiga) [reply deploy/helm/kubecf/BUILD.bazel]: Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/571/commits/5fcab07449d5553739b0266bce81a105f48fa201.

**Pull request merged**